libs/ledger/include/ledger/storage_unit/transaction_store_sync_service.hpp

```diff
@@ -145,6 +147,12 @@ class TransactionStoreSyncService : public TransactionSink
   State OnResolvingObjects();
   State OnTrimCache();
 
+  // Avoid processing transactions that have been recently
+  // seen or are already in storage
+  bool                                   AlreadySeen(chain::Transaction const &tx);
+  std::set<byte_array::ConstByteArray>   recently_seen_txs_;
+  std::deque<byte_array::ConstByteArray> recently_seen_txs_ordered_;
```

Comment:
I think actually that I would wrap this into a object by itself i.e. a SlidingWindowSet or similar? Thoughts?